test/unit/uk/gov/hmrc/exports/services/mapping/AuthorisationHoldersBuilderSpec.scala

```diff
@@ -33,13 +33,41 @@ class AuthorisationHoldersBuilderSpec extends UnitSpec with ExportsDeclarationBu
 
     "build the expected WCO declaration" when {
 
-      "the ExportsDeclaration instance contains no holders" in {
-        val model = aDeclaration(withoutDeclarationHolders())
-        val declaration = new Declaration()
+      "the ExportsDeclaration instance contains no holders" when {
+        "no declaration holders" in {
+          val model = aDeclaration(withoutDeclarationHolders())
+          val declaration = new Declaration()
 
-        authorisationHoldersBuilder.buildThenAdd(model, declaration)
+          authorisationHoldersBuilder.buildThenAdd(model, declaration)
+
+          declaration.getAuthorisationHolder mustBe empty
+        }
+
+        "the user didn't enter any authorisation code and" when {
+          "the user selects a GVMS port on /location-of-goods" should {
```

Comment:
I think its important to mention here in this test line that he user has selected a GVMS GLC but has not supplied the declarant, exporter or representative details (otherwise it seems to contradict the test at line 118)